src/Notifier.tsx

```diff
@@ -176,6 +166,14 @@ export class NotifierRoot extends React.PureComponent<ShowNotificationParams, St
     }).start();
   }
 
+  private setHideTimer() {
+    const { duration = DEFAULT_DURATION } = this.props;
```

Comment:
```suggestion
    const { duration = DEFAULT_DURATION } = this.showParams;
```
Since `this.props` contains only params that were passed to `NotifierRoot` or `NotifierWrapper` components we should use `this.showParams` here that contains `this.props` and all the params that were passed to the `showNotification` function.

Also please move type definition of `duration` (file `types.ts`) from interface `ShowNotificationParams` to interface `ShowParams`

Reply:
```suggestion
    const { duration = DEFAULT_DURATION } = this.showParams ?? {};
```
Also, `showParams` can be `null`, so we should count this case 🙂